src/clj/game/utils.clj

```diff
@@ -136,3 +137,68 @@
                   (when (pred x)
                     idx))
                 coll))
+
+(defn swap!*
+  "Wraps `swap!` to log changes for diffing later."
+  [a f & args]
+  (when (instance? game.core.state.State @a)
+    (if (and
+          (not (:unsent-changes-send-all @a))
+          (or (= assoc f)
+              (= assoc-in f)
+              (= update f)
+              (= update-in f)
+              (= dissoc f)
+              (= dissoc-in f)))
+      (swap! a update :unsent-changes conj (first args))
+      (do
+        (println "Called swap!* on function:" f)
+        (swap! a assoc :unsent-changes-send-all true))))
+  (apply swap! a f args))
+
+(defn deep-select-keys
+  [m in-keys]
+  (loop [ks in-keys
+         res {}]
+    (if (empty? ks)
+      res
+      (if (keyword? (first ks))
+        (recur (rest ks) (assoc res (first ks) (get m (first ks))))
```

Comment:
Make ks [ks (seq in-keys)], then check (if ks ...) then use (next ks) to call seq on ks when recurring.